libs/coin-modules/coin-solana/src/network/chain/web3.ts

```diff
@@ -122,7 +122,10 @@ async function* getTransactionsBatched(
     const transactions = await api.getParsedTransactions(
       signaturesInfoBatch.map(tx => tx.signature),
     );
-    const txsDetails = transactions.reduce((acc, tx, index) => {
+    const sortedTransactions = transactions
+      .filter(tx => tx)
+      .sort((a, b) => (b?.slot ?? 0) - (a?.slot ?? 0));
```

Comment:
You should be able to chain now I think
```suggestion
      .sort((a, b) => (b?.slot ?? 0) - (a?.slot ?? 0))
      .reduce((acc, tx, index) => {
```